tests/NotificationTest.php

```diff
@@ -25,6 +25,26 @@ public function test_send_idempotent_notification()
         $this->assertEquals("/send", $response->path);
     }
 
+    public function test_send_enhanced_notification()
+    {
+        $message = (object) [];
+        $response = $this->getCourierClient()->sendEnhancedNotification($message, NULL);
+
+        $this->assertEquals("POST", $response->method);
+        $this->assertEquals("application/json", $response->content);
+        $this->assertEquals("/send", $response->path);
+    }
+
+    public function test_send_enhanced_idempotent_notification()
+    {
+        $message = (object) [];
+        $response = $this->getCourierClient()->sendEnhancedNotification($message, "idempotent");
+
+        $this->assertEquals("POST", $response->method);
```

Comment:
it would be nice to have a check to make sure `buildIdempotentRequest` was called and not `buildRequest`.

Reply:
I would rather move fast on this change over super exhaustive testing of a deeper if condition. that being said if we know a quick way to spy and test it happy to add that but didn't think it was necessary